test/Microsoft.AspNet.Razor.Test/Parser/CSharp/CSharpBlockTest.cs

```diff
@@ -736,6 +737,275 @@ public void ParsersCanNestRecursively()
                     Factory.Code("        }").AsStatement().Accepts(AcceptedCharacters.None)));
         }
 
+        public static TheoryData BlockWithEscapedTransitionData
+        {
+            get
+            {
+                var factory = CreateDefaultSpanFactory();
+                var datetimeBlock = new ExpressionBlock(
+                    factory.CodeTransition(),
+                    factory.Code("DateTime.Now")
+                        .AsImplicitExpression(CSharpCodeParser.DefaultKeywords)
+                        .Accepts(AcceptedCharacters.NonWhiteSpace));
+
+                return new TheoryData<string, Block>
+                {
+                    {
+                        // Double transition in attribute value
```

Comment:
Good coverage here and in `HtmlDocumentTest`. About the only thing we _might_ want in addition would be `<span foo='@ @' />`.  Closest I see is two comments with a space between them but I may be missing an almost-double-transition case elsewhere.